src/plugins/intel_cpu/src/transformations/snippets/tpp/pass/scalar_to_scalar_tpp.cpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/itt.hpp"
+#include "scalar_to_scalar_tpp.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "snippets/op/scalar.hpp"
+#include "transformations/snippets/tpp/op/scalar.hpp"
+#include "transformations/snippets/tpp/op/modifiers.hpp"
+#include "snippets/lowered/port_connector.hpp"
+
+
+namespace ov {
+namespace intel_cpu {
+namespace tpp {
+namespace pass {
+
+ScalarToScalarTPP::ScalarToScalarTPP() {
+    MATCHER_SCOPE(ScalarToScalarTPP);
+
+    auto snippets_scalar = ov::pass::pattern::wrap_type<ov::snippets::op::Scalar>();
+
+
+    auto callback = [=](ov::pass::pattern::Matcher& m) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "ov::intel_cpu::pass::ScalarToScalarTPP")
+        const auto node = ov::as_type_ptr<ov::snippets::op::Scalar>(m.get_match_root());
+        OPENVINO_ASSERT(node, "Failed to obtain a valid Scalar Op in ScalarToScalarTPP");
+        size_t num_connected_tpp = 0;
+        const auto& target_ins = node->get_output_target_inputs(0);
+        for (const auto& in : target_ins) {
+            if (dynamic_cast<tpp::modifier::TensorProcessingPrimitive*>(in.get_node()))
+                num_connected_tpp++;
+        }
+        if (num_connected_tpp == 0)
+            return false;
+        OPENVINO_ASSERT(num_connected_tpp == target_ins.size(), "Either all or none Scalar outputs should be TPP");
```

Comment:
Theoretically, we can support cases when one scalar has TPP and non-TPP consumers: we can just copy TPP scalar and reassign it only to TPP operations

Reply:
Thanks for the idea! I added it to a comment.
However, I still don't think that we should support it here because in 99% of cases this assert indicates invalid configuration. The thing is that composite subgraphs are typically restricted to a certain unsupported operations (e.g. Convert of PowerStatic) but none of them have Scalar inputs.